### DISCUSSION TITLE
Why Redis database use in limit-count plugin instead of etcd one?

### DISCUSSION THREAD

**Discussion by rakesh-308** [apache/apisix#7931] [General]
Came across this "policy" attribute in limit-count plugin. I wonder why etcd was not used to store persistent limit count information rather than redis as etcd was already there and it is highly consistent and distributed, light weight database

**Comment by tzssangglass** [comment]
limit-count requires high frequency reads and writes to the database, and etcd is not as good as redis in this scenario.

A quick glance at other project implementations shows that redis is also used as the database in cluster limit-count.

**Reply by rakesh-308** [reply]
Make sense but at the same time it increase the overhead to have two databases running in the cluster..just a thought

**Reply by tzssangglass** [reply]
It is a reasonable architectural idea to choose the right components according to the scenario. Instead of ignoring the scenario and using etcd directly because it is a database, this is the wrong architectural thinking. My own thoughts.

**Reply by tokers** [reply]
Another big reason is that, if you want to use ETCD in the `limit-count` plugin, you have to configure a write-able ETCD user for APISIX, which violates the design that we want to make APISIX DP side read-only for ETCD.

**Reply by rakesh-308** [reply]
Fair enough.
